web/src/Web.App/Attributes/TrustAuthorizationAttribute.cs

```diff
@@ -0,0 +1,20 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+namespace Web.App.Attributes;
+
+[AttributeUsage(AttributeTargets.Class)]
+public class TrustAuthorizationAttribute : AuthorizeAttribute, IAuthorizationFilter
+{
+    public void OnAuthorization(AuthorizationFilterContext context)
+    {
+        var companyNumber = context.RouteData.Values["companyNumber"]?.ToString();
+
+        var isValid = context.HttpContext.User.Claims.Any(c => companyNumber != null && c.Type == "Trusts" && c.Value.Contains(companyNumber));
```

Comment:
Do we need null checks on User.Claims or Claim.Value or is this safe enough as-is?

Reply:
Should be enough as-is. But we can ensure we have integration tests to cover this when we fully integrate DSI.